Sources/xcproj/PBXBuildRule.swift

```diff
@@ -0,0 +1,136 @@
+import Foundation
+
+/// A PBXBuildRule is used to specify a method for transforming an input file in to an output file(s).
+final public class PBXBuildRule: PBXObject, Equatable {
+
+    // MARK: - Attributes
+
+    /// Element compiler spec.
+    public var compilerSpec: String
+
+    /// Element file patterns.
+    public var filePatterns: String?
```

Comment:
Out of curiosity. Did you check the optionality of the attributes with Xcode?

Reply:
Thanks for fixing this up and merging. @pepibumur, yes, I tried examining the encoded form of of a number of different permutations. `compilerSpec` is required so specify how the file should be processed, but `filePatterns` is optional since it's a filter on `fileType`.

Reply:
No problem @briantkelley 😛  it was just curiosity. The changes should be in xcproj 1.8.0 ready to be used :)